src/python/pants/backend/project_info/peek.py

```diff
@@ -159,11 +179,44 @@ async def get_target_data(
         for tgt, hs in zip(targets_with_sources, hydrated_sources_per_target)
     }
 
+    family_adaptors = await _get_target_family_and_adaptor_for_dep_rules(
+        *(tgt.address for tgt in sorted_targets),
+        description_of_origin="`peek` goal",
+    )
+    dependencies_rules_map = {
+        tgt.address: describe_ruleset(family.dependencies_rules.get_ruleset(tgt.address, adaptor))
+        for tgt, (family, adaptor) in zip(sorted_targets, family_adaptors)
+        if family.dependencies_rules is not None
+    }
+    dependents_rules_map = {
+        tgt.address: describe_ruleset(family.dependents_rules.get_ruleset(tgt.address, adaptor))
+        for tgt, (family, adaptor) in zip(sorted_targets, family_adaptors)
+        if family.dependents_rules is not None
+    }
+    all_effective_dep_rules = await MultiGet(
```

Comment:
It's odd that we're introducing this new bit of terminology, "effective", but populating it with something using the terminology "application". Why not be consistent, and call this "applied_dependency_rules" or "applicable_dependency_rules"?

Reply:
That said, I'm not sure what the purpose of this is? If a rule bans a dependency then that dependency won't exist, and so there will be no rule that applies to it. So this is just for rules that allow dependencies?

Reply:
It allows you to see what rules are defined, and also, in case you see dependencies you thought would have been denied but aren't, which rules allowed them. So it lets you peek into the defined visibility rules. It's been requested for a way to find out what rules have been defined and what applies to certain targets etc. 
 
Agree on effective vs applicable. I'll change that.

Reply:
I still think this is a bit confusing? What is in effective_ vs in dependencies/dependents?

Reply:
in dependencies/dependents you set up a bunch of rules yea. They may or may not "capture" anything. Applied (effective) rules are those that actually hit something. You've got a dependency relation between two targets where some rule(s) is/are in play.

Reply:
In play with respect to what though? This target as a source? As a dest?

Reply:
as a source, it's between the target and its dependencies

Reply:
Got it, makes sense. So there is some asymmetry here (which I guess is fine).

Reply:
Or, do we want four fields:

- Rules applying to this node as a source
- The subset of these that actually applied for some edge
- Rules applying to this node as a dest
- The subset of these that actually applied for some edge

?

Or something? I guess where this is tripping me up is that the "applied" rules are a property of an *edge*, not of a target.

Reply:
> Rules applying to this node as a dest

is that not very expensive? or, it shouldn't be worse than `pants dependents ..`  ?
but that would be some extra lookup from what we do in peek now, any way.

Reply:
I guess so. I'm just trying to wrap my head around the utility of the "applied" data. Most of the time rules will ban edges, so those edges won't exist. So this is only for the case of an edge that was explicitly allowed by a rule, rather than being allowed by default because no rule bans it? And we're not actually putting this information on the edge, but on the source target, and we're not saying which edge it referred to?

Reply:
There may be rules that use "warn" rather than "deny". We may introduce an option to treat "deny" as "warn", we may also, for `peek` and `lint` not exclude denied edges for the purpose of reporting (https://github.com/pantsbuild/pants/pull/17671#issuecomment-1330773206).

> we're not saying which edge it referred to?

Oh, but we do.

```
src/python/pants/explorer/server/graphql -> src/python/pants : ALLOW
python_sources src/python/pants/explorer/server/graphql/field_types.py -> python_sources src/python/pants/__init__.py"

```

This tells us that for the dependency edge from `../graphql/field_types.py` to `../pants/__init__.py` was allowed (by default, as there where no rules involved.

The other example, for comparison:

```
src/python/pants/explorer/server/graphql -> 3rdparty/python/BUILD[src/python/pants/explorer/server/**] : ALLOW
python_sources src/python/pants/explorer/server/graphql/field_types.py -> python_requirements 3rdparty/python#strawberry-graphql",

```

Edge from `../graphql/field_types.py` to `../python#strawberry-graphql` was allowed by the rule `.../server/**` of the dependents rule defined in `3rdparty/python/BUILD`, and there were no rules in play on the dependency side of the edge.

Reply:
Also worth noting is, that although this data only shows what has been allowed, the logs will hold information about what was denied, so that is already available. And if you expected something to be denied but wasn't, this will help pin point why.

Reply:
Ah, gotcha. OK, that is neat.